cli/args.go

```diff
@@ -423,16 +442,15 @@ func parseMultiStringArg(args []string, argName string, defaultValue []string) (
 	stringArgs := []string{}
 
 	for i, arg := range args {
-		if arg == fmt.Sprintf("--%s", argName) {
-			newOption, deprecated := isDeprecatedOption(argName)
-			if deprecated {
-				util.GlobalFallbackLogEntry.Warnf("Command line option %s is deprecated, please consider using %s", argName, newOption)
-			}
-			if (i + 1) < len(args) {
-				stringArgs = append(stringArgs, args[i+1])
-			} else {
-				return nil, errors.WithStackTrace(ArgMissingValue(argName))
-			}
+		var nextArg *string
+		if (i + 1) < len(args) {
+			nextArg = &args[i+1]
+		}
+		val, hasVal, err := getOptionArgIfMatched(arg, nextArg, argName)
+		if err != nil {
+			return nil, errors.WithStackTrace(ArgMissingValue(argName))
```

Comment:
Why is this line not like `parseIntArg` line 428? `return nil, err`? I think that's the only error that's returned from `getOptionArgIfMatched`, so I would consider using `err` here.

Reply:
Good catch - addressed in https://github.com/gruntwork-io/terragrunt/pull/1923